src/python/pants/backend/docker/goals/run_image.py

```diff
@@ -3,20 +3,32 @@
 
 from __future__ import annotations
 
-from typing import cast
+from dataclasses import dataclass
+from typing import Iterator, cast
 
 from pants.backend.docker.goals.package_image import BuiltDockerImage, DockerFieldSet
 from pants.backend.docker.subsystems.docker_options import DockerOptions
 from pants.backend.docker.util_rules.docker_binary import DockerBinary
 from pants.core.goals.package import BuiltPackage, PackageFieldSet
-from pants.core.goals.run import RunDebugAdapterRequest, RunRequest
+from pants.core.goals.run import RunDebugAdapterRequest, RunFieldSet, RunRequest
 from pants.engine.env_vars import EnvironmentVars, EnvironmentVarsRequest
 from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.engine.unions import UnionRule
+
+
+@dataclass(frozen=True)
+class DockerRunFieldSet(DockerFieldSet, RunFieldSet):
```

Comment:
Oops. This didn't work as well as in my mind, as it turns out I didn't pay attention to the nuance of using a union type in a `Get`, so the exact type of the instance does matter so I can't fool the engine to pass my subclass to the package image rule.

Will have to re-think this one slightly.